src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -506,13 +548,28 @@ func (l *commitLog) write() {
 		l.metrics.success.Inc(numWritesSuccess)
 	}
 
-	writer := l.writerState.writer
-	l.writerState.writer = nil
+	// Ensure that there is no active background goroutine in the middle of reseting
+	// the secondary writer / modifying its state.
+	l.waitForSecondaryWriterAsyncResetComplete()
 
-	l.closeErr <- writer.Close()
+	var multiErr xerrors.MultiError
+	for i, writer := range l.writerState.writers {
+		if writer == nil {
+			// Can be nil in the case where the background goroutine spawned in openWriters
+			// encountered an error trying to re-open it.
```

Comment:
nit: If you really want to you could create a `forEachWriters(func(writer) { ... })` helper function to make sure all for loops does this.

Reply:
Yeah that would be a little cleaner, but I'd rather be explicit and have it crash in places where we don't expect to ever be nil than have it silently do something weird.